Source/objects.cpp

```diff
@@ -1959,6 +1959,8 @@ void OperateBook(Player &player, Object &book)
 		    player.getId(),
 		    0,
 		    0);
+		player.executedSpell.spellType = RSPLTYPE_SPELL;
+		ConsumeSpell(player, SPL_GUARDIAN);

```

Comment:
Yeah, I was struggling with that question myself. There's a good chance this code simply called `UseMana()` as a side-effect of calling `AddMissile()`, and it's entirely possible that it could have consumed a staff charge or scroll as a result indicating that this was probably not intended behavior. Furthermore, the code always casts slvl 0 Guardian from the Bone Chamber book regardless of the player's spell level, which seems to suggest that it shouldn't really be the player's Guardian. I was kinda thinking the mana loss could still be justified as the cost of learning more about Guardian, but in hindsight that intent is obviously not reflected in the code and would be inconsistent with every other spell book in the game.

I suppose I would ultimately agree that the call to `ConsumeSpell()` should be removed here.

Reply:
I agree. This makes more sense and we also got better code. 😉